source-code/sdk-js/src/runtime/runtime.ts

```diff
@@ -0,0 +1,53 @@
+import type { Resource } from "@inlang/core/ast"
+import { BaseLookupFunctionArgs, createLookupFunction, LookupFunction } from "./lookup-function.js"
+
+const dummyResource = {
+	type: "Resource",
+	languageTag: { type: "LanguageTag", name: "en" },
+	body: [],
+} satisfies Resource
+
+const fallbackLookupFunction: LookupFunction = () => ""
+
+export type RuntimeState<Language extends string = string> = {
+	resources: Map<Language, Resource>
+	language: Language
+}
+
+export const initRuntime = <
+	Language extends string,
+	LookupFunctionArgs extends BaseLookupFunctionArgs = BaseLookupFunctionArgs,
+>() => initBaseRuntime<Language, LookupFunctionArgs>()
+
+export const initBaseRuntime = <
+	Language extends string,
+	LookupFunctionArgs extends BaseLookupFunctionArgs = BaseLookupFunctionArgs,
+>(
+	state: RuntimeState<Language> = {
+		resources: new Map(),
+		language: undefined as unknown as Language,
+	},
+) => {
+	const loadResource = async (language: Language) => {
+		const resource = dummyResource // TODO: actually load resource
+		state.resources.set(language, resource)
+	}
+
+	const switchLanguage = (language: Language) => (state.language = language)
+
+	const getLanguage = () => state.language
```

Comment:
what should we do if the language was never set? Return `undefined`? Throw an error?

Reply:
@samuelstroschein @benjaminpreiss what do you thing here?

Reply:
i assume a language always exist, worst case the reference language, right?

Reply:
I think it is best to return `string | undefined` here.

Reply:
Yes. We can always narrow the type in a later iteration.